tests/bsim/bluetooth/samples/compile.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+# Copyright 2023-2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# Compile all the applications needed by the bsim tests in these subfolders
+
+#set -x #uncomment this line for debugging
+set -ue
+
+: "${BSIM_COMPONENTS_PATH:?BSIM_COMPONENTS_PATH must be defined}"
+: "${ZEPHYR_BASE:?ZEPHYR_BASE must be set to point to the zephyr root\
+ directory}"
+
+WORK_DIR="${WORK_DIR:-${ZEPHYR_BASE}/bsim_out}"
+
+BOARD_ROOT="${BOARD_ROOT:-${ZEPHYR_BASE}}"
+
+mkdir -p ${WORK_DIR}
+
+source ${ZEPHYR_BASE}/tests/bsim/compile.source
+
+if [ "${BOARD}" == "nrf5340bsim_nrf5340_cpuapp" ]; then
+  app=samples/bluetooth/peripheral_hr \
+    sysbuild=1 \
+    conf_overlay=overlay-phy_coded.conf \
+    exe_name=bs_${BOARD}_${app}_prj_conf \
```

Comment:
@aescolar Now the name length is too long in the script file, should I reduce the name `central_hr_peripheral_hr` or do you have any suggestion so that the executable name does not execeed the 100 char column limit in the compliance?

Reply:
Oh well.. silly checks
If you want you can replace:
`Execute ./bs_${BOARD}_tests_bsim_bluetooth_samples_central_hr_peripheral_hr_prj_conf_overlay-phy_coded_conf \`
with
`Execute ./bs_${BOARD_TS}_$(guess_test_long_name)_prj_conf_overlay-phy_coded_conf \`
(Note that due to hwmv2 now we need to use BOARD_TS to ensure we don't have "/" in the name.